client/src/api/common/data.ts

```diff
@@ -0,0 +1,31 @@
+import { useState } from "react";
+import { Array, Record, Static, String, Unknown } from "runtypes";
+import { useChannelShaped } from "./socket";
+
+const ModuleDataT = Record({
+    /** Sensor data */
+    sensors: Array(Record({
+        /** Type of data */
+        type: String,
+        /** Value */
+        value: Unknown
+    }))
+});
+
+type _ModuleDataT = Static<typeof ModuleDataT>;
+
+export interface ModuleDataT extends _ModuleDataT {}
```

Comment:
Why not just use `type ModuleDataT = Static<typeof ModuleDataT>;`?

(also, do we need the `T`? Not really in line with our structures but happy if it's reasoned)

Reply:
I was using the `T` to indicate that it was the type of the `ModuleData`. Agree that it's not completely necessary - I think it's common (?) in TS libraries but it's definitely not something we've implemented widely in our code.
3940cdd

> Why not just use type ModuleDataT = Static<typeof ModuleDataT>;?

Mainly for consistency. All of our types so far relating to function props and return types have been using interfaces. Of course, types and interfaces are quite similar, but the main difference is that a type cannot extend an interface (requires union rather than extend) which can be annoying to navigate later.

Reply:
Fair enough, happy with this!